apps/dashboard/src/pages/integrations/components/create-integration-sidebar.tsx

```diff
@@ -0,0 +1,124 @@
+  const providers = novuProviders;
+  const { mutateAsync: createIntegration, isPending } = useCreateIntegration();
+  const { mutateAsync: setPrimaryIntegration, isPending: isSettingPrimary } = useSetPrimaryIntegration();
+  const { integrations } = useFetchIntegrations();
+
+  const { selectedIntegration, step, searchQuery, onIntegrationSelect, onBack } = useSidebarNavigationManager({
+    isOpened,
+  });
+
+  const { integrationsByChannel } = useIntegrationList(providers, searchQuery);
+  const provider = providers?.find((p) => p.id === selectedIntegration);
+  const {
+    isPrimaryModalOpen,
+    setIsPrimaryModalOpen,
+    pendingData,
+    handleSubmitWithPrimaryCheck,
+    handlePrimaryConfirm,
+    existingPrimaryIntegration,
+    isChannelSupportPrimary,
+  } = useIntegrationPrimaryModal({
+    onSubmit,
+    integrations,
+    channel: provider?.channel,
+    mode: 'create',
+  });
+  async function onSubmit(data: IntegrationFormData) {
+    if (!provider) return;
+
+    try {
+      const integration = await createIntegration({
+        providerId: provider.id,
+        channel: provider.channel,
+        credentials: data.credentials,
+        name: data.name,
+        identifier: data.identifier,
+        active: data.active,
+        _environmentId: data.environmentId,
+      });
+
+      if (data.primary && isChannelSupportPrimary && data.active) {
+        await setPrimaryIntegration({ integrationId: integration.data._id });
+      }
+
+      onClose();
+    } catch (error: any) {
+      handleIntegrationError(error, 'create');
+    }
+  }
+
+  return (
+    <>
+      <IntegrationSheet
+        isOpened={isOpened}
+        onClose={onClose}
+        provider={provider}
+        mode="create"
+        step={step}
+        onBack={onBack}
+      >
+        {step === 'select' ? (
+          <div className="scrollbar-custom flex-1 overflow-y-auto">
+            <ChannelTabs
+              integrationsByChannel={integrationsByChannel}
+              searchQuery={searchQuery}
+              onIntegrationSelect={onIntegrationSelect}
+            />
+          </div>
+        ) : provider ? (
+          <>
+            <div className="scrollbar-custom flex-1 overflow-y-auto">
+              <IntegrationConfiguration
+                isChannelSupportPrimary={isChannelSupportPrimary}
+                provider={provider}
+                onSubmit={handleSubmitWithPrimaryCheck}
+                mode="create"
+              />
+            </div>
+            <div className="bg-background flex justify-end gap-2 border-t p-3">
+              <Button
+                type="submit"
+                form="integration-configuration-form"
+                isLoading={isPending || isSettingPrimary}
+                size="sm"
+              >
+                Create Integration
+              </Button>
+            </div>
+          </>
+        ) : null}
+      </IntegrationSheet>
+
+      <SelectPrimaryIntegrationModal
```

Comment:
When I switch off the primary flag we should ask to pick a new primary the ones from active integrations, similarly to what we had in v1.

https://github.com/user-attachments/assets/e4d8c677-93d1-410a-954a-d57403478c0c